src/GloryDuckTables2019Lkl.cc

```diff
@@ -0,0 +1,403 @@
+//
+// Disable SetUnitsOfG, since units are used internally by the 
+// class and should not be changed.
+//
+void GloryDuckTables2019Lkl::SetUnitsOfG(Double_t units)
+{
+  cout<< "GloryDuckTables2019Lkl::SetUnitsOfG (" << GetName() << ") Warning: cannot set units for GloryDuckTables2019Lkl." << endl;
+}
+
+////////////////////////////////////////////////////////////////
+//
+// Read the Glory Duck input file and add the 
+// ParabolaLkl describing the -2logL function vs <sv> 
+//
+// Return 0 in case of success, 1 otherwise
+//
+Int_t GloryDuckTables2019Lkl::ReadGloryDuckInputData(TString filename)
+{
+  // try to open file
+  ifstream ff(filename.Data());
+  if(!ff.is_open())
+    {
+      cout << "GloryDuckTables2019Lkl::ReadGloryDuckInputData (" << GetName() << ") Error: could not open file " << filename << endl;
+      return 1;
+    }
+  else
+    cout << "GloryDuckTables2019Lkl::ReadGloryDuckInputData (" << GetName() << ") Message: Reading masses and <sigmav> values from file " << filename << endl;
+
+  vector<Double_t> vsigma ;    // vector with <sv> values
+  vector<Double_t> vlkl ;      // vecot with -2logL values
+
+  // get first line of the file (with <sv> values)
+  string sigmavLine;
+  getline(ff,sigmavLine);
+
+  // read first line of the file
+  std::istringstream row(sigmavLine);
+  Double_t field;
+  while (row>> field)
+    {
+      vsigma.push_back(field);
+    }
+
+  Double_t readingMass = 0.;
+  Double_t readingLkl = 0.;
+  Int_t counter = 0;
+
+  // get mass (first column)
+  while(ff >> readingMass)
+    {
+      // get rest of the line (<sv> values)
+      string LklLine;
+      getline(ff,LklLine);
+
+      // read rest of the line
+      std::istringstream line(LklLine);
+      while (line >> readingLkl)
+        {
+          vlkl.push_back(readingLkl);
+        }
+
+      // add new parabola
+      CreateAndAddNewParabola(readingMass,vsigma[0],vsigma[vsigma.size()-1],vsigma.size(),vsigma.data(),vlkl.data());
+
+      // check if line was complete or not
+      if(vlkl.size() != vsigma.size())
+        {
+          cout << "You have a different line size length for line " << counter+1 << " ! Number of sigmav values = " << vsigma.size() << " while number of likelihood values = " << vlkl.size() << endl; 
+        }
+      counter++;
+      vlkl.clear();
+   }
+
+  ff.close(); 
+
+  SetChecked(kFALSE);
+
+  cout << "GloryDuckTables2019Lkl::ReadGloryDuckInputData (" << GetName() << ") Message: Read " <<  GetSampleArray()->GetEntries() << " parabolas for a total of " << fNMasses << " DM masses." << endl;
+  return 0;
+}
+
+////////////////////////////////////////////////////////////////
+//
+// Add a new ParabolaLkl object representing the -2logL values vs
+// <sv> for one DM mass
+//
+// Return 0 in case of success, 1 otherwise
+//
+Int_t GloryDuckTables2019Lkl::CreateAndAddNewParabola(Double_t mass,Double_t svmin,Double_t svmax,Int_t npoints,Double_t* sv,Double_t* logL) 
```

Comment:
Definitely, svmin and svmax are not needed, they can always be read from sv, which is stored in every parabola